### PR TITLE
Better warnings when cross compiler setup fails

### PR DESCRIPTION
**Pull request by jpakkane**

### CODE REVIEW

File: mesonbuild/environment.py [hunk 1]
```diff
@@ -448,6 +448,12 @@ def get_gnu_compiler_type(defines):
             return GCC_CYGWIN
         return GCC_STANDARD
 
+    def warn_about_lang_pointing_to_cross(self, compiler_exe, evar):
+        evar_str = os.environ.get(evar, 'WHO_WOULD_CALL_THEIR_COMPILER_WITH_THIS_NAME')
+        if evar_str == compiler_exe:
```
Comment (jon-turney) [inline mesonbuild/environment.py]: Why not?

```
evar_str = os.environ.get(evar, None)
if evar_str and evar_str == compiler_exe:
```
Or perhaps a comment explaining why that isn't right?

File: mesonbuild/environment.py [hunk 2]
```diff
@@ -448,6 +448,12 @@ def get_gnu_compiler_type(defines):
...
+            mlog.warning('''Env var %s seems to point to the cross compiler.
```
Comment (jon-turney) [inline mesonbuild/environment.py]: Perhaps 'environment variaible'  in full to use actual words?

**Comment by jon-turney** [comment]
Maybe this needs drawing out a bit more in the documentation somewhere, since this is a difference of expectations vs. autotools (where CC will be the host machine compiler, and something like CC_FOR_BUILD will be invented for the build machine compiler)

**Comment by jpakkane** [comment]
Documentation updated.

**Pull request merged**